app/components/work_version_metadata_component.rb

```diff
@@ -39,12 +39,18 @@ class WorkVersionMetadataComponent < ActionView::Component::Base
   ].freeze
 
   def initialize(work_version:, mini: false)
-    @work_version = work_version
+    @work_version = decorate(work_version)
     @mini = mini
   end
 
   private
 
+    def decorate(work_ver)
+      return work_ver if work_ver.is_a? ResourceDecorator
+
+      ResourceDecorator.new(work_ver)
+    end
```

Comment:
Ditto for showing collections... Dashboard::WorkVersionDecorator inherits from ResourceDecorator, so we should be able to the same decorator throughout.